### PR TITLE
(#208) Site Alert tests

### LINKED ISSUE #208: Redesign:  Site Alerts
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: Slim site alert ( not automated)
  Given user is navigating to "/"
  Then slim site alert is displayed
  And alert link "link" has an href "href"
 And alert is not collapsible or dismissible
```
```gherkin
Scenario: Standard site alert
  Given user is navigating to "/"
  Then standard site alert is displayed
And alert has "dismiss" button
When user clicks on "expand" arrow button
Then alert is "expanded"
And the following links are exposed
|linkText| href|
|.    |.    |
When user clicks on "collapse" arrow button
Then alert is "collapsed"
And alert links are hidden
```

```gherkin
Scenario: Analytics for site alert expand click
  Given user is navigating to "/"
  Then standard site alert is displayed
When user clicks on "expand" arrow button
 Then page click request is sent
And the following parameters should be captured
            | parameter | value                                               |
            | prop4     | D=pev1                                              |
            | prop67    | D=pageName                                          |
            | prop52    | PreHeader\|Standard Alert\|Expand\|Expand |
            | prop68    | PreHeader                                           |
            | prop8     | english                                             |
            | evar2     | D=c8                                                |
            | evar52    | D=c52                                               |
            | evar68    | D=c68                                               |
            | pageURL   | https://{CANONICAL_HOST}/                           |
            | pageName  | {CANONICAL_HOST}/                                   |
            | event63   |                                                     |
            | channel   | NCI Homepage                                        |
            | pev2      | PreHeader:LinkClick                                 |
            | linkType  | lnk_o                                               |
  
Scenario: Analytics for site alert link click
  Given user is navigating to "/"
  Then standard site alert is displayed
When user clicks on "a link" alert link
 Then page click request is sent
And the following parameters should be captured
            | parameter | value                                               |
            | prop4     | D=pev1                                              |
            | prop67    | D=pageName                                          |
            | prop52    | PreHeader\|Slim Alert\|Link Click\|a link |
            | prop68    | PreHeader                                           |
            | prop8     | english                                             |
            | evar2     | D=c8                                                |
            | evar52    | D=c52                                               |
            | evar68    | D=c68                                               |
            | pageURL   | https://{CANONICAL_HOST}/                           |
            | pageName  | {CANONICAL_HOST}/                                   |
            | event63   |                                                     |
            | channel   | NCI Homepage                                        |
            | pev2      | PreHeader:LinkClick                                 |
            | linkType  | lnk_o                                               |
            | link      | a link                                             |

Scenario: Analytics for site alert dismiss click
  Given user is navigating to "/"
  Then standard site alert is displayed
When user clicks on "expand" arrow button
 Then page click request is sent
And the following parameters should be captured
            | parameter | value                                               |
            | prop4     | D=pev1                                              |
            | prop67    | D=pageName                                          |
            | prop52    | PreHeader\|Standard Alert\|Dismiss\|Dismiss |
            | prop68    | PreHeader                                           |
            | prop8     | english                                             |
            | evar2     | D=c8                                                |
            | evar52    | D=c52                                               |
            | evar68    | D=c68                                               |
            | pageURL   | https://{CANONICAL_HOST}/                           |
            | pageName  | {CANONICAL_HOST}/                                   |
            | event63   |                                                     |
            | channel   | NCI Homepage                                        |
            | pev2      | PreHeader:LinkClick                                 |
            | linkType  | lnk_o                                               |
      

Scenario: After dismissing site alert, it is no longer visible
  Given user is navigating to "/"
  Then standard site alert is not displayed
```
### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by kate-mashkina**
- added after hook to
remove site alert
- added tests
Closes #208

### CODE REVIEW

**Comment by kate-mashkina** [comment]
Screen Shot 2022-10-05 at 6 00 02 PM

**Comment by sarinapadilla** [comment]
This PR looks good to me!

File: cypress/e2e/SiteAlert/StandardAlert.feature [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            | pev2      | PreHeader:LinkClick                         |
+            | linkType  | lnk_o                                       |
+        When user is navigating to "/about-cancer/coping/feelings"
+        Then standard site alert is not displayed
+
+    Scenario: login to remove alert (so after hook doesn't have to)
+        Given user is navigating to "/user/login"
+        When user enters credentials
+        And user clicks "Log in" button
+        When user is logged in and the user name "admin" is displayed in the toolbar
```
Comment (bryanpizzillo) [inline cypress/e2e/SiteAlert/StandardAlert.feature:159]: Does this seem to end abruptly, i.e. it is not deleted?

File: cypress/e2e/SiteAlert/StandardAlert.feature [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        And alert has "Dismiss alert" button
+        When user clicks on "expand" arrow button
+        Then alert is "expanded"
+        And the following links are exposed
+            | linkText                                          | href |
+            | What people with cancer should know               | #    |
+            | Get the latest public health information from CDC | #    |
+        When user clicks on "collapse" arrow button
+        Then alert is "collapsed"
+        And alert links are hidden
```
Comment (bryanpizzillo) [inline cypress/e2e/SiteAlert/StandardAlert.feature:58]: Wait -- the alert defaults to being collapsed.

Can you please add a test for expanding then moving to another page and seeing it is still expanded? Then collapse it and move to another page and seeing it is still collapsed.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

File: cypress/e2e/SiteAlert/StandardAlert.feature (hunk 1 above)
Comment (KateMashkinaNIH) [reply cypress/e2e/SiteAlert/StandardAlert.feature:159]: this is just logging in for the after hook, so I don't have to duplicate the login code in a hook)

File: cypress/e2e/SiteAlert/StandardAlert.feature (hunk 2 above)
Comment (KateMashkinaNIH) [reply cypress/e2e/SiteAlert/StandardAlert.feature:60]: added steps to verify alert state

**Pull request merged**